app/uk/gov/hmrc/gform/submission/SubmissionRef.scala

```diff
@@ -16,42 +16,68 @@
 
 package uk.gov.hmrc.gform.submission
 
+import java.math.BigInteger
+import java.security.MessageDigest
+
 import cats.data.State
 import play.api.libs.json._
 import uk.gov.hmrc.gform.sharedmodel.ValueClassFormat
 import uk.gov.hmrc.gform.sharedmodel.form.EnvelopeId
-import uk.gov.hmrc.gform.typeclasses.Rnd
 
+import scala.math.pow
 import scala.util.Random
 
 case class SubmissionRef(value: String) extends AnyVal {
   override def toString = value
 }
 
 object SubmissionRef {
+  val radix = 36
+  val digits = 11
+  val comb: Stream[Int] = Stream.continually(List(1, 3)).flatten
 
   val oformat: OFormat[SubmissionRef] = ValueClassFormat.oformat("submissionRef", SubmissionRef.apply, _.value)
   val vformat: Format[SubmissionRef] =
     ValueClassFormat.vformat("submissionRef", SubmissionRef.apply, x => JsString(x.value))
 
-  def random(implicit rnd: Rnd[Random]) = createSubmissionRef(rnd())
+  def apply(value: EnvelopeId): SubmissionRef = SubmissionRef(getSubmissionReference(value))
 
-  def createSubmissionRef(r: Random): SubmissionRef = {
+  def createSubmissionRef(envelopeId: EnvelopeId): SubmissionRef =
+    SubmissionRef(getSubmissionReference(envelopeId).replace("-", ""))
 
-    def getChars(i: Int) = State[Random, String](r => (r, alphanumeric(r).take(i).toList.mkString))
+  def getSubmissionReference(envelopeId: EnvelopeId): String =
```

Comment:
I think that all of the methods in the companion object should be private, with the exception of apply